packages/flutter_tools/lib/src/web/chrome.dart

```diff
@@ -241,6 +239,67 @@ class ChromiumLauncher {
     ), skipCheck);
   }
 
+  Future<Process> _spawnChromiumProcess(List<String> args) async {
+    // Keep attempting to launch the browser until one of:
+    // - Chrome launched successfully, in which case we just return from the loop.
+    // - The tool detected an unretriable Chrome error, in which case we throw ToolExit.
+    while (true) {
+      final Process process = await _processManager.start(args);
+
+      bool exited = false;
+      unawaited(process.exitCode.whenComplete(() {
+        exited = true;
+      }));
+
+      process.stdout
+        .transform(utf8.decoder)
+        .transform(const LineSplitter())
+        .listen((String line) {
+          _logger.printTrace('[CHROME]: $line');
+        });
+
+      // Wait until the DevTools are listening before trying to connect. This is
+      // only required for flutter_test --platform=chrome and not flutter run.
+      bool hitGlibcBug = false;
+      await process.stderr
```

Comment:
This can throw dart:io exceptions if the process fails in certain ways/at certain times. Ditto for `process.stdout` which is another reason to do `.listen().asFuture()` and await the `Future` in a `try {} catch {}` for it.

Reply:
This sounds like a good future enhancement. I'd prefer to wait until it actually happens, so we can recreate the exact situation in a test. I ran this code 18000 times and it worked fine.